### PR TITLE
Plugins

### PR DESCRIPTION
**Pull request by ar2r13**
# Plugins
- ## $mol_view_ghost
    In $mol we have a $mol_view_ghost components. It like $mol_view but it not creates an additional wrap in HTML. 

    #### Problems of $mol_view_ghost
    - In view.tree we must wrap our component, it is inconvenient
    - It broke ```id```'s in HTML

- ## Solution is plugins /
    ```plugins /``` is an array of plugins. Plugin is a component which can be supplemented with the logic of the current components.
    #### For example
    In the example we create a list with navigation (using $mol_nav)
    - ###### $mol_view_ghost
        ``` 
        $my_list $mol_nav
            keys_y <= list_options /
            Sub <= List $mol_list
                rows <= list_options /
        ```
    - ###### plugins
        ```
        $my_list $mol_list
            plugins / 
                <= Nav $mol_nav
                    keys_y <= list_options /
            rows <= list_options /
        ```

# View.Tree Right-side binding ```=>```
In view.tree we have several binding types
- Left-side binding ```<=```
- Double-side binding ```<=>```

I think we need yet one type

- Right-side binding ```=>```

This type will return actual value of an entity with which it works.
It is ideally suited for plugins, when we need the actual value of the plugin, but do not pass it a value.
    
# $mol_meter
Is a brand new plugin which returns an actual size and offsets of component.
```
    $my_button $mol_button
        plugins /
            <= Meter $mol_meter
                width => cur_width
                height => cur_height 
                top => cur_top
                bottom => cur_bottom
                left => cur_left
                right => cur_right
        title <= label @ /I know my size

### CODE REVIEW

File: meter/meter.view.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+namespace $.$mol {
+	export class $mol_meter extends $.$mol_meter {
+		
+		request_id = 0;
```
Comment (nin-jin) [inline meter/meter.view.ts]: Приватные поля нужно начинать с подчёркивания.

File: select/select.view.tree [hunk 2]
```diff
@@ -19,27 +19,28 @@ $mol_select $mol_view
 		sub <= clear_option_content /
 			<= clear_option_message @ \Clear
 			<= Clear_icon $mol_icon_cross
-	sub / 
+	plugins /
```
Comment (nin-jin) [inline select/select.view.tree]: У нас везде свойства в единственном числе: field, event, style.

File: view/readme.md [hunk 3]
```diff
@@ -50,6 +50,19 @@ Returns dictionary of event handlers. The event handlers are bind to the DOM-ele
...
+It is an array of plugins. Plugin is a component which can be supplemented with the logic of the current components.
+For example
+
+In the example we create a list with navigation (using $mol_nav)
+```
+$my_list $mol_nav
+    keys_y <= list_options /
+    Sub <= List $mol_list
+        rows <= list_options /
+```
```
Comment (nin-jin) [inline view/readme.md]: А при чём тут плагины?

File: view/readme.md [hunk 4]
```diff
@@ -325,9 +338,12 @@ $my_greeter $mol_view
 		<= Input $mol_string
 			hint \Name
 			value?val <=> name?val \
+			enabled?val <= input_enabled - 
 		<= Output $mol_view
 			sub /
 				name?val \
+		<= Check $mol_check
+			checked => input_enabled
```
Comment (nin-jin) [inline view/readme.md]: Странный пример, тут лучше бы подошёл двусторонний биндинг.

File: view/readme.md (hunk 3 above)
Comment (ar2r13) [inline view/readme.md]: Не тот код вставил))

File: view/readme.md (hunk 4 above)
Comment (ar2r13) [inline view/readme.md]: Исправил пример

**Pull request merged**